### PR TITLE
Document limitation with reusable workflows

### LINKED ISSUE #22746: Document limitation with reusable workflows
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations

### What part(s) of the article would you like to see updated?

The [Limitations](https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations) section.

If you create a reusable workflow in repository A and that workflow uses a composite action defined also in repository A, you cannot use the reusable workflow from repository B.

This [community discussion](https://github.com/orgs/community/discussions/18601) does a good job illustrating the issue, but basically the upstream workflow assumes that the composite action will be in the context of repository B's files when it should be looking in repository A's files.

The error you receive in repository B's actions output is similar to the below, which I copied from the linked discussion thread:

> Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/runner/work/CLIENT/REPO/generate-docker-tags@master'. Did you forget to run actions/checkout before running your local action?

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by unchris**
### Why:

Closes https://github.com/github/docs/issues/22746

If you create a reusable workflow in repository A and that workflow uses a composite action defined also in repository A, you cannot use the reusable workflow from repository B. 

This [community discussion](https://github.com/orgs/community/discussions/18601) does a good job illustrating the issue, but basically the upstream workflow assumes that the composite action will be in the context of repository B's files when it should be looking in repository A's files.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

I'm adding a limitation to the Limitations section for Reusable Workflows.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by NGUYENDAIBAO — COMMENTED** [review]
`.`

**Comment by cmwilson21** [comment]
@unchris Thank you for opening a PR and linking it to your issue! ✨ 

We appreciate all the info. I'll get this triaged for review! ⚡

File: content/actions/using-workflows/reusing-workflows.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Called workflows that are owned by the same user or organization{% ifversion ghe
 {% ifversion private-actions %}{% else %}* Reusable workflows stored within a private repository can only be used by workflows within the same repository.{% endif %}
 * Any environment variables set in an `env` context defined at the workflow level in the caller workflow are not propagated to the called workflow. For more information about the `env` context, see "[Context and expression syntax for GitHub Actions](/actions/reference/context-and-expression-syntax-for-github-actions#env-context)."{% ifversion actions-reusable-workflow-matrix %}{% else %}
 * The `strategy` property is not supported in any job that calls a reusable workflow.{% endif %}
+* You cannot use a reusable workflow from another repository if it uses a composite action.
```
Comment (hubwriter) [inline content/actions/using-workflows/reusing-workflows.md:92]:
```suggestion
* You cannot use a reusable workflow from another repository if it uses a composite action. For more information, see "[About custom actions](/actions/creating-actions/about-custom-actions#composite-actions)."
```

**Review by hubwriter — DISMISSED** [review]
@unchris - Many thanks for making this docs update. 👍 

I'll get this merged & published.
🚀

**Review by hubwriter — APPROVED** [review]

**Pull request merged**